### ISSUE TITLE
Errors when rendering with batch cameras

### ISSUE THREAD

**Issue by itswyri** [facebookresearch/pytorch3d#652]
I found a bug when rendering images with multiple cameras (batch_sized cameras).
I fixed it on my own. But I'd like to know if it is correct and want to share it with you.

## 🐛 Bugs / Unexpected behaviors
error occurs at `pytorch3d/renderer/blending.py` line 187 (or 188) `z_inv = (zfar - fragments.zbuf) / (zfar - znear) * mask`

`zfar`'s shape at dimension -1 doesn't coincide with its `fragments.zbuf`

## Instructions To Reproduce the Issue:
```python
# camera
        R, T = look_at_view_transform(2.7, 0., 0.) # (1, 3, 3), (1, 3)
        cameras = FoVOrthographicCameras(device=device, R=R, T=T, znear=0.05, zfar=1000)

        # light
        lights = PointLights(device=device, location=[[0, -1, 1]])

        # materials
        materials = Materials(device=device,
                              specular_color=[color],
                              shininess=0.0)

        # Rasterization settings for differentiable rendering, where the blur_radius
        # initialization is based on Liu et al, 'Soft Rasterizer"
        sigma = 1e-4
        raster_settings_soft = RasterizationSettings(
            image_size=resolution[::-1],
            blur_radius=np.log(1. / 1e-4 - 1.) * sigma,
            faces_per_pixel=50,
            perspective_correct=False
        )

        # Differentiable soft renderer using per vertex RGB colors for texture
        renderer = MeshRenderer(
            rasterizer=MeshRasterizer(
                cameras=cameras,
                raster_settings=raster_settings_soft,
            ),
            shader=SoftPhongShader(
                device=device,
                cameras=cameras,
                lights=lights,
                materials=materials
            )
        )
        



        scale_xyz = torch.stack([sx, sy, torch.ones_like(sx)], dim=1)

        R_, T_ = look_at_view_transform(2.7, 0., 0.)
        R = R_.expand(verts.shape[0], -1, -1)
        T = T_.expand(verts.shape[0], -1)
        cameras = FoVOrthographicCameras(device=device, R=R, T=T, znear=self.znear, zfar=self.zfar, max_x=max_x, min_x=min_x, max_y=max_y, min_y=min_y, scale_xyz=scale_xyz)
        # max_x, min_x, max_y, min_y : (N) sized tensor with different values


        textures = TexturesVertex(verts_features=torch.ones_like(verts))

        meshes = Meshes(verts=verts,
                        faces=faces,
                        textures=textures)
        images = self.renderer(meshes, cameras=cameras)
```

### How I fixed it
When doing an element-wise operation, PyTorch tries to duplicate the shape 1 dimension of one tensor to the shape of the other tensor (>1). The matching order seems last dimension first. In this case, `zfar` has shape (N) and `fragments.zbuf` has shape (N, ..., 50) (the first dimension's shape is batch_size, and the last dimension's shape is faces_per_pixel).
According to broadcasting rule, it occurs errors. So I add shape reshaping code prior to the `z_inv` computation at line 187
```
    if isinstance(zfar, torch.Tensor):
        if zfar.ndim > 0:
            if zfar.shape[0] > 1:
                zfar = zfar.reshape(*((zfar.shape[0],) + (1,)*(fragments.zbuf.ndim - 1)))
    if isinstance(znear, torch.Tensor):
        if znear.ndim > 0:
            if znear.shape[0] > 1:
                znear = znear.reshape(*((znear.shape[0],) + (1,)*(fragments.zbuf.ndim - 1)))

    z_inv = (zfar - fragments.zbuf) / (zfar - znear) * mask
```
```

**Comment by bottler** [comment]
I think this is the same as #550

**Comment by nikhilaravi** [comment]
Thanks for raising this issue @itswyri! We will fix it!

**Comment by nikhilaravi** [comment]
This issue should now be fixed by https://github.com/facebookresearch/pytorch3d/commit/bc8361fa471e39280b1c5e3717309167b08ffed5.
